src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +362,194 @@ namespace cryptonote
+        {
+          // request non-mempool txs
+          NOTIFY_REQUEST_FLUFFY_MISSING_TX::request missing_tx_req;
+          missing_tx_req.b = arg.b;
+          missing_tx_req.hop = arg.hop;
+          BOOST_FOREACH(auto& tx_hash, need_tx_hashes)
+          {
+            missing_tx_req.missing_tx_hashes.push_back(tx_hash);
+          }
+          
+          post_notify<NOTIFY_REQUEST_FLUFFY_MISSING_TX>(missing_tx_req, context);
+        }
+        else // whoo-hoo we've got em all ..
+        {
+          block_complete_entry b;
+          b.block = arg.b.block;
+          b.txs = have_tx;
+          
+          std::list<block_complete_entry> blocks;
+          blocks.push_back(b);
+          m_core.prepare_handle_incoming_blocks(blocks);
+
+          for(auto tx_blob_it = arg.b.txs.begin(); tx_blob_it!=arg.b.txs.end();tx_blob_it++)
+          {
+            cryptonote::tx_verification_context tvc = AUTO_VAL_INIT(tvc);
+            m_core.handle_incoming_tx(*tx_blob_it, tvc, true, true);
+            if(tvc.m_verifivation_failed)
+            {
+              LOG_PRINT_CCONTEXT_L1("Block verification failed: transaction verification failed, dropping connection");
+              m_p2p->drop_connection(context);
+              m_core.cleanup_handle_incoming_blocks();
+              m_core.resume_mine();
+              return 1;
+            }
+          }
+          
+          block_verification_context bvc = boost::value_initialized<block_verification_context>();
+          m_core.handle_incoming_block(arg.b.block, bvc); // got block from handle_notify_new_block
+          m_core.cleanup_handle_incoming_blocks(true);
+          m_core.resume_mine();
+          
+          if( bvc.m_verifivation_failed )
+          {
+            LOG_PRINT_CCONTEXT_L0("Block verification failed, dropping connection");
+            m_p2p->drop_connection(context);
+            return 1;
+          }
+          if( bvc.m_added_to_main_chain )
+          {
+            ++arg.hop;
+            //TODO: Add here announce protocol usage
+            relay_fluffy_block(arg, context);
+          }
+          else if( bvc.m_marked_as_orphaned )
+          {
+            context.m_state = cryptonote_connection_context::state_synchronizing;
+            NOTIFY_REQUEST_CHAIN::request r = boost::value_initialized<NOTIFY_REQUEST_CHAIN::request>();
+            m_core.get_short_chain_history(r.block_ids);
+            LOG_PRINT_CCONTEXT_L2("-->>NOTIFY_REQUEST_CHAIN: m_block_ids.size()=" << r.block_ids.size() );
+            post_notify<NOTIFY_REQUEST_CHAIN>(r, context);
+          }            
+        }
+      } 
+      else
+      {
+        LOG_ERROR_CCONTEXT
+        (
+          "sent wrong block: failed to parse and validate block: \r\n"
+          << epee::string_tools::buff_to_hex_nodelimer(arg.b.block) 
+          << "\r\n dropping connection"
+        );
+        
+        m_core.resume_mine();
+        m_p2p->drop_connection(context);        
```

Comment:
Yea, I'll change things around to conform to the already existing coding style (gotta start following that contributor's guide) :)